File: frontend/lib/core/models/payment.dart
class Payment {
  const Payment({
    required this.id,
    required this.assignmentId,
    required this.amount,
    required this.currency,
    required this.status,
    this.intentId,
  });

  factory Payment.fromJson(Map<String, dynamic> json) => Payment(
        id: json['id'] as String,
        assignmentId: json['assignment_id'] as String? ?? '',
        amount: json['amount'] as int? ?? 0,
        currency: json['currency'] as String? ?? 'JPY',
        status: json['status'] as String? ?? 'requires_payment_method',
        intentId: json['stripe_payment_intent_id'] as String?,
      );

  final String id;
  final String assignmentId;
  final int amount;
  final String currency;
  final String status;
  final String? intentId;
}

File: backend/schemas/payment.py
from enum import Enum
from typing import Optional

from pydantic import BaseModel, EmailStr, Field

from .base import PaginatedResponse, TimestampedModel


class PaymentStatus(str, Enum):
    REQUIRES_PAYMENT_METHOD = "requires_payment_method"
    REQUIRES_CONFIRMATION = "requires_confirmation"
    REQUIRES_ACTION = "requires_action"
    PROCESSING = "processing"
    SUCCEEDED = "succeeded"
    CANCELED = "canceled"


class PaymentBase(TimestampedModel):
    assignment_id: str
    amount: int = Field(..., ge=100)
    currency: str = Field(default="jpy", max_length=3)
    stripe_payment_intent_id: Optional[str] = None
    stripe_transfer_id: Optional[str] = None
    status: PaymentStatus = PaymentStatus.REQUIRES_PAYMENT_METHOD
    metadata: Optional[dict] = None


class PaymentCreate(BaseModel):
    assignment_id: str
    amount: int = Field(..., ge=100)
    currency: str = Field(default="jpy", max_length=3)


class PaymentUpdate(BaseModel):
    status: Optional[PaymentStatus] = None
    stripe_payment_intent_id: Optional[str] = None
    stripe_transfer_id: Optional[str] = None
    metadata: Optional[dict] = None


class PaymentRead(PaymentBase):
    id: str


class PaymentList(PaginatedResponse[PaymentRead]):
    pass


class ConnectAccountRequest(BaseModel):
    email: EmailStr


class PaymentIntentCreateRequest(BaseModel):
    amount: int = Field(..., ge=100)
    worker_stripe_account: str
